stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/handler/NotificationHandler.kt

```diff
@@ -72,4 +73,11 @@ public interface NotificationHandler {
      * Dismiss all notifications.
      */
     public fun dismissAllNotifications()
+
+    /**
+     * Handles [android.Manifest.permission.POST_NOTIFICATIONS] permission lifecycle.
+     *
+     * @param status Represents current [android.Manifest.permission.POST_NOTIFICATIONS] permission status.
+     */
+    public fun onNotificationPermissionStatus(status: NotificationPermissionStatus) { /* no-op */ }
```

Comment:
I would consider removing the default implementation from the interface and moving it to the handler. I know we don't have to implement it in all cases but it will be easier to find this method if someone wants to implement their own handler